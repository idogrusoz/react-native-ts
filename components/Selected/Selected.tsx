import React, { FunctionComponent } from "react";
import { TouchableOpacity, StyleSheet } from "react-native";
import { useDispatch } from "react-redux";
import { removeSelected } from "../../redux/actions/actions";
import { View, Text } from "../Themed";

interface SelectedProps {
    name: string;
    index: number;
}

const Selected: FunctionComponent<SelectedProps> = ({ name, index }) => {
    const dispatch = useDispatch();
    const handleDelete = () => {
        dispatch(removeSelected(index));
    };
    return (
        <View style={styles.wrapper}>
            <Text>{name}</Text>
            <TouchableOpacity style={styles.delete} onPress={handleDelete}>
                <Text>x</Text>
            </TouchableOpacity>
        </View>
    );
};

const styles = StyleSheet.create({
    wrapper: {
        display: "flex",
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "center",
        height: 25,
        paddingHorizontal: 30,
        marginVertical: 10,
    },
    delete: {
        display: "flex",
        flexDirection: "row",
        justifyContent: "center",
        height: 20,
        width: 30,
    },
});

export default Selected;
