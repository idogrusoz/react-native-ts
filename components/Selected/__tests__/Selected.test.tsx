import React from "react";
import { fireEvent, render, RenderAPI } from "@testing-library/react-native";
import Selected from "../Selected";
const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
    useDispatch: () => mockDispatch,
}));

describe("Selected test", () => {
    let wrapper: RenderAPI;
    beforeEach(() => {
        wrapper = render(<Selected name="name" index={1} />);
    });

    it("renders without crash", () => {
        expect(wrapper).toMatchSnapshot();
    });
    it("dispays the name", () => {
        const text = wrapper.getByText("name");
        expect(text).not.toBeNull();
    });
    it("calls dispatch", () => {
        const touch = wrapper.getByText("x");
        fireEvent(touch, "onPress");
        expect(mockDispatch).toBeCalled();
    });
});
