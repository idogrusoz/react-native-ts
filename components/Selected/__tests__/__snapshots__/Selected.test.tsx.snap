// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selected test renders without crash 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#b2ebf2",
        "display": "flex",
        "flexDirection": "row",
        "height": "5%",
        "justifyContent": "space-between",
        "marginVertical": "2%",
        "paddingHorizontal": "3%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        undefined,
      ]
    }
  >
    name
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          undefined,
        ]
      }
    >
      x
    </Text>
  </View>
</View>
`;
